### DISCUSSION TITLE
Renovate rules_python's requirements_lock.txt

### DISCUSSION THREAD

**Discussion by lalten** [mvukov/rules_ros2#134] [General]
This is not directly related to rules_ros2, but since you're playing around with Mend Renovate: I am wondering how to make it work with the requirements_lock.txt so that Python modules are upgraded (and tested in CI) automatically. The issue is that renovate does not understand how to run `//:requirements.update` and instead produces diffs like https://github.com/lalten/rules_appimage/pull/51/files

**Comment by mvukov** [comment]
I'm thinking to set up a scheduled CI workflow for this; e.g. run every two weeks. The workflow/job would run bazel pip resolver and create a new PR if there are changes to the requirements lock file.

**Reply by lalten** [reply]
I guess that works when you don't pin the entries in the requirements.in file. Pinning the .in file I think makes sense as it allows to be more specific when upgrading just a specific dependency. It limits the number of surprises caused by upgrading other deps.
